File: src/main/java/de/fluxparticle/animation/signal/SignalDoubleLine.java
package de.fluxparticle.animation.signal;

public class SignalDoubleLine extends SignalPath<Double> {

    public static class Builder extends SignalPath.Builder<Double> {

        @Override
        public SignalPath<Double> build() {
            return new SignalDoubleLine(start, end);
        }

    }

	public SignalDoubleLine(Double start, Double end) {
		super(start, end);
	}

	@Override
	public Double apply(Double t) {
		return start * (1.0 - t) + end * t;
	}

}

File: src/main/kotlin/de/fluxparticle/animation/signal/SignalDoubleLine.kt
package de.fluxparticle.animation.signal

class SignalDoubleLine(start: Double, end: Double) : SignalPath<Double>(start, end) {

    class Builder : SignalPath.Builder<Double>() {

        override fun build(): SignalPath<Double> {
            return SignalDoubleLine(start, end)
        }

    }

    override fun apply(t: Double): Double {
        return start * (1.0 - t) + end * t
    }

}
